Feature replace native mapdrive

bammab opened the pull request:
This commit replace the code to be used to mount the workbench netshare on windows. Instead of using a native c++ library, this patch call the API directly by ctypes methods inside of python without the workaround of the external lib.

I searched for all files where these MapDrive.dll exists, but I'm not sure, if I catched all descriptions and scripts.

Commit:
replace native cpp mapdrive lib by ctype function calls to API inside python

Commit:
switch back to allocate by function

Use the same call as the mapdrive.dll before. Switch from fixed size string
buffer to wide char pointer.

pandel commented:
Hey @bammab !

Just a short reaction regarding your both PRs: MANY THANKS! I am currently not at work and am unable to update the repo, but I promise to update its content with v8.4.4 shortly, so we can check your PRs against the current release.

Please be patient, it may take about a week...

Regards,
Holger

bammab commented:
No problem! Have nice holidays ;-)

If the v8.4.4 is online, I'll rebase both PRs onto the updated master. I don't know if Github will do this automatically or I need to rebase the second PR than one more time (because the master and feature-branch have the same base at the moment)...

Best regards from Osnabrück

Commit:
replace FormatMessage by python ctypes.FormatError

pandel commented:
Hey Mattias!

Released latest version 8.4.5 to the public and pushed all changes to Github. Now we are on the same spot :-) Perhaps you can test you PRs in advance, but I come back in the next few days and have a deeper look at your changes.

Grüße aus Moers :-)

bammab commented:
Hey Holger,

I've rebased all my feature branches and my master (used in the other PR) against your current master without hassle.
Please feel free to check the code. As said before - if a second rebase is required after the first accepted PR I'll do that asap.
Also it's possible that I create a single squash commit If you don't want to see the single steps.

This PR has a additional commit compared to the first version. I didn't know, that python has an integrated method to call the FormatMessage API of windows until I study more of the ctypes docu. So I removed the API code and used this one. This patch reduce the mapdrive.py to 93 lines, just 2 lines more as before, but without the complete c++ library.

pandel commented:
Many, many thanks for all your changes! I appreciate all of them, especially as I have no time atm to make any improvements myself!
